nnvm/src/top/tensor/reduce.cc

```diff
@@ -70,7 +70,7 @@ inline TShape ReduceShapeImpl(const TShape& ishape,
   if (keepdims) {
```

Comment:
suggest to add `CHECK(r_axes.ndim() < indim)` before this line to make the code more self-explained.